arch/arm/src/armv7-m/arm_saveusercontext.S

```diff
@@ -52,7 +56,11 @@
 
 	.globl	up_saveusercontext
 	.globl	up_saveusercontext
+#ifdef __ghs__
+	.type	up_saveusercontext, $function
+#else
 	.type	up_saveusercontext, %function
```

Comment:
Why not just use double quotes to solve the common compiler issue?
`	.type	up_saveusercontext, "function"`

Reply:
> Why not just use double quotes to solve the common compiler issue? ` .type up_saveusercontext, "function"`

I have tried this in local, this cannot fix the build error, and will report the following error:
CC: sched/sched_waitpid.c [asarm] (error https://github.com/apache/nuttx/pull/2230) common/gnu/fork.S 80: bad directive
.type up_fork , "function"

Reply:
OK